### PR TITLE
Development

### PR DESCRIPTION
**Pull request by cgbriggs99**
I made the changes requested, as well as a test file to run some tests.

### CODE REVIEW

**Comment by cgbriggs99** [comment]
I added a custom gradient finder, and a test file with a couple tests. The tests I am currently running the code against are a linear regression with no bias, a linear regression with bias, and a 2-d linear regression with bias.

**Comment by cgbriggs99** [comment]
I implemented some of the changes you suggested. Everything is now in subplots, there are command line arguments, I moved many of the extra function arguments to kwargs, the scripts are now in the scripts directory, added return descriptions to the docstrings, and removed most of the return values. If I missed anything, let me know.

File: scripts/mdr.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        grad = gradient(loss, pred, w, [step for i in range(len(w_start))],
+                        x, y)
+        abs_grad = [abs(g) for g in grad]
+        if min(abs_grad) < grad_conv :
+            if printout :
+                print(f"Converged in {it} iterations.")
+                print(f"w value: {w}")
+            return w
+        else :
+            # Use the Barzilai-Borwein method to have a variable step size.
```
Comment (bgpeyton) [inline scripts/mdr.py]: This method has been shown to be effective for not just any gradient descent, but _stochastic_ gradient descent as well, which is the method we would ultimately want to use for bigger regression problems. Great inclusion!

File: scripts/test.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    # Find how many rows and columns of subplots to show.
+    plot_r = 2 if plot_total > 3 else 1
+    plot_c = math.ceil(plot_total / plot_r)
+    figure = plt.figure()
+
+    # Whether or not to print debug information.
+    if kwargs["print_debug"] == "on" :
+        print("Test case 0:")
+        w0 = mdr.cross_validation(input0, output0, 3, pred0, l2_square,
+                                  [0], print=True)
```
[the remark this answers is not in the document]
Comment (cgbriggs99) [reply scripts/test.py]: It is. I just wasn't paying attention.

File: scripts/test.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+        plot_n += 1
+    elif kwargs["non_lin_2d"] == "contour" :
+        ax = figure.add_subplot(plot_r, plot_c, plot_n)
+        ax.contourf(X, Y, Z, cmap = cm.coolwarm)
+        ax.title(f"Fifth Test: 2-D Non-linear r² = {r2(output4, ys)}")
+        plot_n += 1
+
+    plt.show()
+
+if __name__ == "__main__" :
```
[the remark this answers is not in the document]
Comment (cgbriggs99) [reply scripts/test.py]: Sure.

File: scripts/mdr.py (hunk 1 above)
Comment (cgbriggs99) [reply scripts/mdr.py]: Thanks!

**Comment by bgpeyton** [comment]
LGTM. Going to merge this in, and update the README to detail the additional features later.

**Pull request merged**